tests/unit/models/search.js

```diff
@@ -0,0 +1,287 @@
+        eissns: ["0082-3974"]
+      }
+    };
+
+    expect(search.getEndpoint(scope)).toContain("search?issns=00284793");
+  });
+
+  it("should select the eissn when the journal search result has no issn", function() {
+    var scope = {
+      document: {
+        content_type: "Journal",
+        eissns: ["0082-3974"]
+      }
+    };
+
+    expect(search.getEndpoint(scope)).toContain("search?issns=00823974");
+  });
+
+  it("should build an article endpoint for an article search result", function() {
+    var scope = {
+      document: {
+        content_type: "Journal Article",
+        dois: ["10.1136/bmj.h2575"]
+      }
+    };
+
+    expect(search.getEndpoint(scope)).toContain("articles/doi/10.1136%2Fbmj.h2575");
+  });
+
+  it("should build an article endpoint for an article search result and include its journal", function() {
+    var scope = {
+      document: {
+        content_type: "Journal Article",
+        dois: ["10.1136/bmj.h2575"]
+      }
+    };
+
+    expect(search.getEndpoint(scope)).toContain("articles/doi/10.1136%2Fbmj.h2575?include=journal");
+  });
+
+  it("should include a browzineWebLink in the BrowZine API response for a journal", function() {
+    var data = search.getData(journalResponse);
+    expect(data).toBeDefined();
+    expect(search.getBrowZineWebLink(data)).toEqual("https://browzine.com/libraries/XXX/journals/10292");
+  });
+
+  it("should include a browzineWebLink in the BrowZine API response for an article", function() {
+    var data = search.getData(articleResponse);
+    expect(data).toBeDefined();
+    expect(search.getBrowZineWebLink(data)).toEqual("https://browzine.com/libraries/XXX/journals/18126/issues/7764583?showArticleInContext=doi:10.1136/bmj.h2575");
+  });
+
+  it("should include a coverImageUrl in the BrowZine API response for a journal", function() {
+    var data = search.getData(journalResponse);
+    expect(data).toBeDefined();
+    expect(search.getCoverImageUrl(data, journalResponse)).toEqual("https://assets.thirdiron.com/images/covers/0028-4793.png");
+  });
+
+  it("should include a coverImageUrl in the BrowZine API response for an article", function() {
+    var data = search.getData(articleResponse);
+    expect(data).toBeDefined();
+    expect(search.getCoverImageUrl(data, articleResponse)).toEqual("https://assets.thirdiron.com/images/covers/0959-8138.png");
+  });
+
+  it("should not build a coverImageUrl when the BrowZine API response is missing the journal data type", function() {
+    var data = search.getData(journalResponse);
+    delete data.type;
+    expect(data).toBeDefined();
+    expect(search.getCoverImageUrl(data, journalResponse)).toBeNull();
+  });
+
+  it("should not build a coverImageUrl when the BrowZine API response is missing the article data type", function() {
+    var data = search.getData(articleResponse);
+    delete data.type;
+    expect(data).toBeDefined();
+    expect(search.getCoverImageUrl(data, journalResponse)).toBeNull();
+  });
```

Comment:
Excellent job getting all the setup logic into a shared `beforeEach`, it makes the rest of these tests very short and readable and the use cases very apparent